BHoM_Engine/Compute/SwitchOffRecording.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2024, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base.Debugging;
+using BH.oM.Base.Attributes;
+using System.Linq;
+using System.ComponentModel;
+using System;
+
+namespace BH.Engine.Base
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Switch off the entire logging system used within BHoM. Any part of the code base which tries to log any note, warning, error, or other event into the log system will be housed in the silent recorder and not displayed to users. By default, all recording systems are turned on when BHoM is initialised.")]
+        [Input("areYouSure", "Set this to true if you are sure you want to turn off all log recording. This boolean exists so that if this component is placed on a BHoM UI accidently, it does not turn off the system unless then expressly requested.")]
+        public static void SwitchOffRecording(bool areYouSure)
```

Comment:
I have a though on this PR as a whole. We are relying on the user to remember to turn back on the logging system the use of code where this is implemented could have effects where during an entire session the logging has been turned off. 

I'm considering an approach where we use the "using" statement.
We can create a class that implements IDisposable, called EventsSupressor, and an enum.
So by using "using" we only allow suppression between a scope and the suppression is removed once the scope has been exceeded. This makes it so that you can only have your suppression last for the lifetime of your method, rather than affecting the whole system after your method has been called, you you've forgotten to renable

``` csharp
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

main();

void main (params string[] args){
    Supression suppressionConfig = Supression.Warnings & Supression.Errors & Supression.Notes;

    using(var eventSupressor = new EventSupressor(suppressionConfig)){
        // do everything you want to do
    }

   // outside of the scope the supression has been returned to normal 
}

enum Supression{
    Warnings,
    Errors,
    Notes
}


class EventSupressor : IDisposable
{
    List<string> m_SupressedLog = new List<string>();

    public EventSupressor(Supression config)
    {
      //logic that turns off supresses 
        if(config.HasFlag(Supression.Warnings)){
           //supress warnings 
        }

        if(config.HasFlag(Supression.Errors)){
           //supress errors
        }

        if(config.HasFlag(Supression.Notes)){
           //supress Notes
        }
    }

    public void Dispose()
    {
      //logic that turns returns logging to proper state
      System.Console.WriteLine("Logging Supression has been ended.");
    }

```